### DISCUSSION TITLE
Allow renaming files in archives

### DISCUSSION THREAD

**Discussion by PetbkA** [doublecmd/doublecmd#1192] [Suggestions]
> Double Commander
> Version: **1.1.0 beta**
> Revision: 32
> Commit: **cd3a69c65**
> Build date: 2023/08/05
> Lazarus: 2.2.7.0
> FPC: 3.2.2
> Platform: **x86_64-Win64-win32/win64**
> OS version: **Windows 10 Home 22H2 x86_64**

Can't rename files inside e.g. _zip_ archive while having the ability do delete and add them. I suggest to support renanimg files too

**Comment by alexx2000** [comment]
[WCX-API](https://ghisler.github.io/WCX-SDK/table_of_contents.htm) does not support rename operation.

**Comment by ags1234** [comment]
+1
I could also use this feature

**Comment by PetbkA** [comment]
> [WCX-API](https://ghisler.github.io/WCX-SDK/table_of_contents.htm) does not support rename operation.

But _Total Commander 10.00 x64_ can rename files inside at least _zip_s (and can't inside _tar_s). Although may be Chrisitan Ghisler wanted to leave it as TC's unique feature 🤷‍♂️

**Comment by alexx2000** [comment]
It is because TC has built-in .zip support (not via plugin). Technically I can extend WCX-API myself of course.

**Reply by ags1234** [reply]
Also build support for moving files in archive, since the checkbox is already there.
See my bug report here: https://github.com/doublecmd/doublecmd/issues/601
